cmd/subnetcmd/upgradecmd/apply.go

```diff
@@ -204,29 +200,151 @@ func applyLocalNetworkUpgrade(subnetName string, sc models.Sidecar) error {
+		ux.Logger.PrintToUser("2. Create a directory with the blockchainID in the configured chain-config-dir (e.g. $HOME/.avalanchego/chains/%s) if doesn't already exist.", blockchainIDstr)
+		ux.Logger.PrintToUser("3. Create an `upgrade.json` file in the blockchain directory with the content of your upgrade file.")
+		upgr, err := app.ReadUpgradeFile(subnetName)
+		if err == nil {
+			var prettyJSON bytes.Buffer
+			if err := json.Indent(&prettyJSON, upgr, "", "    "); err == nil {
+				ux.Logger.PrintToUser("   This is the content of your upgrade file as configured in this tool:")
+				fmt.Println(prettyJSON.String())
+			}
 		}
-		jsonBytes, err := json.Marshal(wrapper)
+		fmt.Println()
+		ux.Logger.PrintToUser("   *************************************************************************************************************")
+		ux.Logger.PrintToUser("   * Upgrades are tricky. The syntactic correctness of the upgrade file is important.                          *")
+		ux.Logger.PrintToUser("   * The sequence of upgrades must be strictly observed.                                                       *")
+		ux.Logger.PrintToUser("   * Make sure you understand https://docs.avax.network/nodes/maintain/chain-config-flags#subnet-chain-configs *")
+		ux.Logger.PrintToUser("   * before applying upgrades manually.                                                                        *")
+		ux.Logger.PrintToUser("   *************************************************************************************************************")
+		return nil
+	}
+	_, _, err := validateUpgrade(subnetName, networkKey, sc)
+	if err != nil {
+		return err
+	}
+
+	ux.Logger.PrintToUser("The chain config dir avalanchego uses is set at %s", avalanchegoChainConfigDir)
+	// give the user the chance to check if they indeed want to use the default
+	if avalanchegoChainConfigDir == avalanchegoChainConfigDirDefault {
+		useDefault, err := app.Prompt.CaptureYesNo("It is set to the default. Is that correct?")
 		if err != nil {
-			app.Log.Debug("failed to marshaling upgrades lock file content", zap.Error(err))
-			return nil
+			return err
 		}
-		if err := app.WriteLockUpgradeFile(subnetName, jsonBytes); err != nil {
-			app.Log.Debug("failed to write upgrades lock file", zap.Error(err))
+		if !useDefault {
+			avalanchegoChainConfigDir, err = app.Prompt.CaptureExistingFilepath(
+				"Enter the path to your custom chain config dir (*without* the blockchain ID, e.g /my/configs/dir)")
+			if err != nil {
+				return err
+			}
 		}
+	}
 
-		return nil
+	ux.Logger.PrintToUser("Trying to install the upgrade files at the provided %s path", avalanchegoChainConfigDir)
+	chainDir := filepath.Join(avalanchegoChainConfigDir, sc.Networks[networkKey].BlockchainID.String())
+	destPath := filepath.Join(chainDir, constants.UpgradeBytesFileName)
+	if err = os.Mkdir(chainDir, constants.DefaultPerms755); err != nil && !os.IsExist(err) {
```

Comment:
does Mkdir expand env variables? Is the default path containing env variables?

Reply:
Good catch. I am changing this right away in the flag configuration.